mono/mini/mini-amd64.c

```diff
@@ -1294,6 +1294,168 @@ mono_arch_get_allocatable_int_vars (MonoCompile *cfg)
 	return vars;
 }
 
+
+#ifdef TARGET_MACH
+static int tls_gs_offset;
+#endif
+
+static gboolean
+mono_amd64_have_fast_tls (void)
+{
+#ifdef TARGET_MACH
+	static gboolean have_fast_tls = FALSE;
+	static gboolean inited = FALSE;
+	guint8 *ins;
+
+	if (inited)
+		return have_fast_tls;
+
+	ins = (guint8*)pthread_getspecific;
+
+	/*
+	 * We're looking for these two instructions:
+	 *
+	 * mov    %gs:[offset](,%rdi,8),%rax
+	 * retq
+	 */
+	have_fast_tls = ins [0] == 0x65 &&
```

Comment:
Could we replace these magic numbers with constants or add comments to make this easier to change in the future if necessary? E.g.:

```
65 // GS:
48 // REX.W
8B // MOV
04 // RAX
FD // RDI*8+
```